airflow/providers/google/marketing_platform/hooks/display_video.py

```diff
@@ -127,3 +144,77 @@ def run_query(self, query_id: str, params: Dict[str, Any]) -> None:
             .runquery(queryId=query_id, body=params)
             .execute(num_retries=self.num_retries)
         )
+
+    def create_sdf_download_operation(self, body_request: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Creates an SDF Download Task and Returns an Operation.
+        :param body_request: Body request.
+        :type body_request: Dict[str, Any]
+
+        More information about body request n be found here:
+        https://developers.google.com/display-video/api/reference/rest/v1/sdfdownloadtasks/create
+
+        """
+        request = (
+            self.get_conn_to_display_video()  # pylint: disable=no-member
+            .sdfdownloadtasks()
+            .create(body_request=body_request)
+            .execute(num_retries=self.num_retries)
+        )
+        return request["name"]
+
+    def get_sdf_download_operation(self, name: Dict[str, Any]):
+        """
+        Gets the latest state of an asynchronous SDF download task operation.
+        :param name: The name of the operation resource.
+        :type name: str
+        """
+
+        request = (
+            self.get_conn_to_display_video()  # pylint: disable=no-member
+            .sdfdownloadtasks()
+            .operation()
+            .get(name=name)
+            .execute(num_retries=self.num_retries)
+        )
+        return request
+
+    def download_media(self, resource_name: str, filename: str):
+        """
+        Downloads media.
+        :param resource_name:  of the media that is being downloaded.
+        :type resource_name: str
+        :param filename: object used for saving result of operation get_sdf_download_tasks
+        """
+
+        request = (
+            self.get_conn_to_display_video()  # pylint: disable=no-member
+            .media()
+            .download_media(resource_name=resource_name)
+        )
+        with open(filename, "w+") as target_file:
+            downloader = MediaIoBaseDownload(target_file, request, chunksize=1024 * 1024)
+            done = False
+            while done is False:
+                status, done = downloader.next_chunk()
+                if status:
+                    self.log.info("Download %d.", int(status.progress() * 100))
+            self.log.info("Download Complete!")
```

Comment:
Does this operation requires `flush`?

Reply:
Hm, I am not 100% sure. Here is the example https://github.com/googleapis/google-api-python-client/blob/master/docs/media.md of handling it.
@mik-laj wdyt?

Reply:
From what I see in campaign manager ops flush is required:
```
        with tempfile.NamedTemporaryFile() as temp_file:
            downloader = http.MediaIoBaseDownload(
                fd=temp_file, request=request, chunksize=self.chunk_size
            )
            download_finished = False
            while not download_finished:
                _, download_finished = downloader.next_chunk()

            temp_file.flush()
            # Upload the local file to bucket
            gcs_hook.upload(
                bucket_name=self.bucket_name,
                object_name=report_name,
                gzip=self.gzip,
                filename=temp_file.name,
                mime_type="text/csv",
            )
```
without that empty file is uploaded to GCS. That's how I've learned about flush...

Reply:
Convinced :)